qualtran/resource_counting/_success_prob.py

```diff
@@ -0,0 +1,49 @@
+#  Copyright 2024 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import logging
+from typing import Callable
+
+from attrs import frozen
+
+from qualtran import Bloq
+
+from ._call_graph import get_bloq_callee_counts
+from ._costing import CostKey
+
+logger = logging.getLogger(__name__)
```

Comment:
yes, I'm using logging in the cost framework. It can be very helpful to trace what's going on. I have a branch that adds some logging to the quimb simulation as well. By default, it won't print anything. you have to set up a logger e.g. with `logging.basicConfig(level=logging.INFO)`